SofaKernel/framework/sofa/defaulttype/Mat.h

```diff
@@ -324,12 +324,40 @@ class Mat : public helper::fixed_array<VecNoInit<C,real>,L>
         return id;
     }
 
+
+    template<int S>
+    static bool canSelfTranspose(const Mat<S, S, real>& lhs, const Mat<S, S, real>& rhs)
+    {
+        return &lhs == &rhs;
+    }
+
+    template<int I, int J>
+    static bool canSelfTranspose(const Mat<I, J, real>& /*lhs*/, const Mat<J, I, real>& /*rhs*/)
+    {
+        return false;
+    }
+
     /// Set matrix as the transpose of m.
     void transpose(const Mat<C,L,real> &m)
     {
-        for (int i=0; i<L; i++)
-            for (int j=0; j<C; j++)
-                this->elems[i][j]=m[j][i];
+        if (canSelfTranspose(*this, m))
+        {
+            for (int i=0; i<L; i++)
```

Comment:
replacing this block with `this->transpose()`?

Reply:
Do you mean replacing the nested for-loop in 
`if (canSelfTranspose(*this, m))`
by a call to` this->transpose()`?
This would cause the static-assert to trigger, hence the suggestion from  @maxime-tournier to use templated canSelfTranspose() functions.

Reply:
ok. good to me!